core/src/test/java/org/elasticsearch/search/aggregations/bucket/adjacency/InternalAdjacencyMatrixTests.java

```diff
@@ -37,22 +39,24 @@
     @Override
     public void setUp() throws Exception {
         super.setUp();
-        keys = new ArrayList<>();
+        List<String> listOfKeys = new ArrayList<>();
         int numFilters = randomIntBetween(2, 4);
         String[] filters = new String[numFilters];
         for (int i = 0; i < numFilters; i++) {
             filters[i] = randomAlphaOfLength(5);
         }
         for (int i = 0; i < filters.length; i++) {
-            keys.add(filters[i]);
+            listOfKeys.add(filters[i]);
             for (int j = i + 1; j < filters.length; j++) {
                 if (filters[i].compareTo(filters[j]) <= 0) {
-                    keys.add(filters[i] + "&" + filters[j]);
+                    listOfKeys.add(filters[i] + "&" + filters[j]);
                 } else {
-                    keys.add(filters[j] + "&" + filters[i]);
+                    listOfKeys.add(filters[j] + "&" + filters[i]);
                 }
             }
         }
+        Collections.shuffle(listOfKeys, random());
+        keys = Collections.unmodifiableList(listOfKeys.stream().limit(maxNumberOfBuckets()).collect(Collectors.toList()));
```

Comment:
See above comment, I'd like it if we didn't have to truncate the list of keys here.